src/event/event.service.ts

```diff
@@ -259,6 +257,35 @@ export class EventService {
     return paginate(eventQuery, { page, limit });
   }
 
+  async searchAllEvents(
+    pagination: PaginationDto,
+    query: HomeQuery,
+  ): Promise<any> {
+    await this.getTenantSpecificEventRepository();
+    const { page, limit } = pagination;
+    const { search } = query;
+    const eventQuery = this.eventRepository
+      .createQueryBuilder('event')
+      .leftJoinAndSelect('event.user', 'user')
+      .leftJoinAndSelect('event.categories', 'categories')
+      .leftJoinAndSelect('event.group', 'group')
+      .leftJoinAndSelect('event.attendees', 'attendees')
+      .where('event.status = :status', { status: EventStatus.Published })
+      .andWhere('event.visibility = :visibility', {
+        visibility: EventVisibility.Public,
+      });
+
+    if (search) {
+      eventQuery.andWhere(
+        `(event.name LIKE :search OR
+          event.slug LIKE :search)`,
```

Comment:
As you know 'slug' index is already present, though user isn't searching for a slug but by 'name' instead. 
This functionality should include creation of 'name' index on event/group